EIPS/eip-6956.md

```diff
@@ -0,0 +1,515 @@
+
+- MUST implement `transferAnchor(attestation)`, `burnAnchor(attestation)`, `approveAnchor(attestation)` which
+  - MUST use the `decodeAttestationIfValid(attestation)` to determine `to`, `anchor` and `attestationHash`
+  - MUST record each `attestation` used to authorize each token transfer. RECOMMENDED by storing each used `attestationHash`
+  - MUST increment `attestationsUsedByAnchor[anchor]`, whenever an associated token has been transferred through this method
+  - MUST emit `AttestationUsed`
+  - `transferAnchor(attestation)`, corresponding to ERC-721 `safeTransferFrom(from, to, tokenId)` and also responsible for minting further
+    - MUST temporarily set `anchorIsReleased[anchor]=true` to allow a transfer or mint
+    - MUST ensure `anchorIsReleased[anchor]` has the same state as before `transferAnchor()` has been invoked.
+    - MUST either
+      - call `_safeTransferFrom(ownerOf(tokenByAnchor[anchor]), to, tokenByAnchor[anchor])` when `tokenByAnchor[anchor]` exists
+      - or create a new token mapping the `ANCHOR` through calling the de-facto standard `_safeMint(to, newTokenId)` if `tokenByAnchor[anchor]` does not exist. It is RECOMMENDED use the ERC-721-Enumerable mechanics to acquire `newTokenId`.
+    - MUST emit `AnchorTransfer(from, to, anchor, tokenByAnchor[anchor])`
+  - burnAnchor(attestation), corresponding to ERC-721 `burn(tokenId)`
+    - TODO, see reference IMPL in the meantime
+  - approveAnchor(attestation), corresponding to ERC-721 `approve(to, tokenId)`
+    - TODO, see reference IMPL in the meantime
+
+- MUST implement ERC-721 `burn()`
+
+- RECOMMENDED to have a `tokenURI(tokenId)` implemented to return an anchorBased-URI, i.e. `baseURI/anchor`. (= Anchoring metadata to anchored ASSET). Before an anchor is not used for the first time, the ANCHOR's mapping to tokenId is unknown. Hence, using the anchor instead of the tokenId is preferred.
+
+- RECOMMENDED to implement any or multiple of the following interfaces: transferable(tokenId), isSoulbound(tokenId), isNonTransferable (), `isNonTransferable(tokenId)` according to [ERC-6454](eip-6454.md) ...
+
+- MAY implement the `IERC6956AttestationLimited` interface subject to [the proposed AttestationLimited extension](#ierc6956attestationlimited)
+-- This is a MUST when transaction-costs are provided through a central account, e.g. through the ORACLE (or associated authorities) itself to avoid fund-draining.
+
+- MAY implement the `IERC6956Floatable` interface subject to [the proposed Floatable extension](#ierc6956floatable).
+
+### `IERC6956AttestationLimited`
+
+WIP and needs discussion!!!
+
+Every contract compliant to this standard MAY implement the [proposed AttestationLimited interface](../assets/eip-6956/contracts/IERC6956AttestationLimited.sol) and MUST implement ERC-721 and ERC-165 interfaces (subject to “caveats” below):
+
+```Solidity
```

Comment:
````suggestion
```solidity
````

Reply:
done in https://github.com/ethereum/EIPs/commit/9d188578511cb53c00ebe0910043ab50fb40a248